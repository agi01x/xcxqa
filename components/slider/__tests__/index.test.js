import React from 'react';
import { render, mount } from 'enzyme';
import Slider from '..';

describe('Slider', () => {
  it('should show tooltip when hovering slider handler', () => {
    const wrapper = mount(
      <Slider defaultValue={30} />
    );
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseEnter');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseLeave');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });
});
