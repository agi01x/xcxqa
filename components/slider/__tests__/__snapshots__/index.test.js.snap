// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should show tooltip when hovering slider handler 1`] = `
<div>
  <div
    class="ant-tooltip  ant-tooltip-placement-top "
  >
    <div
      class="ant-tooltip-content"
    >
      <div
        class="ant-tooltip-arrow"
      />
      <div
        class="ant-tooltip-inner"
        role="tooltip"
      >
        30
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider should show tooltip when hovering slider handler 2`] = `
<div>
  <div
    class="ant-tooltip  ant-tooltip-placement-top  ant-tooltip-hidden"
  >
    <div
      class="ant-tooltip-content"
    >
      <div
        class="ant-tooltip-arrow"
      />
      <div
        class="ant-tooltip-inner"
        role="tooltip"
      >
        30
      </div>
    </div>
  </div>
</div>
`;
